extensions/mssql/src/main.ts

```diff
@@ -136,14 +136,14 @@ export async function activate(context: vscode.ExtensionContext): Promise<MssqlE
 
 	azdata.ui.registerModelViewProvider('bdc-endpoints', async (view) => {
 
-		const endpointsArray: Array<Utils.IEndpoint> = view.serverInfo.options['clusterEndpoints'];
-		const managementProxyEp = view.serverInfo.options['clusterEndpoints'].find(e => e.serviceName === 'management-proxy');
+		const endpointsArray: Array<Utils.IEndpoint> = Object.assign([], view.serverInfo.options['clusterEndpoints']);
```

Comment:
I'm sorry, I didn't realize this was an array. This won't work for that will it? Have you tested with these changes to verify this? I'd expect the options won't pass through using this. Instead slice or a similar function should let you shallow copy an array so you can modify it. Again, we need to null check if the array even exists first.

Reply:
yes, I tested it and they worked. But I missed the point of null check, I'll add that.